src/scripts/eicrecon-this.sh.in

```diff
@@ -22,7 +22,10 @@ fi
 #----------------- ROOT
 if [[ -f @ROOT_BINDIR@/thisroot.sh ]] ; then
     unset ROOTSYS  # prevent ROOT from removing /usr/local/bin from PATH
-    source @ROOT_BINDIR@/thisroot.sh
+    # zsh and latest eic-shell require "cd where/root/is" before calling ". bin/thisroot.sh"
+    pushd @ROOT_BINDIR@ >/dev/null
+    source thisroot.sh
+    popd >/dev/null
```

Comment:
This comment is an indication that something is wrong in your environment; not necessarily something that needs to get fixed in this script, I think.

Looking at [`thisroot.sh`](https://github.com/root-project/root/blob/master/config/thisroot.sh), you only get a message referring to `where/root/is` if `thisroot.sh` could not determine which shell you are running. You are in a situation where `$SOURCE` remains empty after the [logic](https://github.com/root-project/root/blob/master/config/thisroot.sh#L194-L208), and you are in a directory where neither `bin/thisroot.sh` nor `./thisroot.sh` exists.

Testing in eic-shell in singularity and in docker (both on linux), this doesn't seem to happen, not when calling `source thisroot.sh` nor `source eicrecon-this.sh`. Can you report a reproducer as an issue for this before we fix things that aren't issues, or worse fix things that just hide the underlying issue better?

Reply:
Would be interesting to see output of the failing `set -x; cd /; source /bla/bin/thisroot.sh`.

Reply:
It's hard to see how the change by itself could be harmful, especially because it's almost certainly needed if you compile outside eic-shell in a native zsh. But you raise a good point about hiding underlying issues. I'll work on a reproducer, I can't get a quick one on linux, and for Mac docker, I'll need a fresh docker pull and a bit of time to minimize it (In my setup, it shows when building inside drich-dev, too many steps if I can avoid it)